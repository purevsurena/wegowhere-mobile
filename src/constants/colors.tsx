import { ColorTypes } from '@/types/colors';

const primary = '#4AD8DA';
const dark = '#000000';
const semiTransparentBlack = 'rgba(0, 0, 0, 0.4981)';
const mediumGray = '#8F8F8F';
const lightGray = '#E6E3E6';
const white = '#ffffff';
const transparent = 'transparent';

const colors: ColorTypes = {
  primary,
  dark,
  semiTransparentBlack,
  mediumGray,
  lightGray,
  white,
  transparent,
};

export default colors;
