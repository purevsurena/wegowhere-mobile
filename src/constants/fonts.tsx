import { Platform } from 'react-native';

// FONT FAMILY
const FC_SUBJECT_REGULAR = Platform.select({
  ios: 'FC Subject [Non-commercial] Reg',
  android: 'FC-Subject-Regular',
});
const FC_SUBJECT_BOLD = Platform.select({
  ios: 'FC Subject [Non-commercial] Bd',
  android: 'FC-Subject-Bold',
});

// FONT WEIGHT
const FONT_WEIGHT_400 = '400';
const FONT_WEIGHT_500 = '500';
const FONT_WEIGHT_700 = '700';

const FONT_STYLE_NORMAL = 'normal';

// Heading constant size
const Heading1 = {
  fontSize: 18,
  fontStyle: FONT_STYLE_NORMAL,
  fontFamily: FC_SUBJECT_BOLD,
  fontWeight: FONT_WEIGHT_500,
  lineHeight: 25.2,
};

// H2
const Heading2 = {
  fontSize: 17,
  fontStyle: FONT_STYLE_NORMAL,
  fontFamily: FC_SUBJECT_BOLD,
  fontWeight: FONT_WEIGHT_700,
  lineHeight: 23.8,
};

// H3
const Heading3 = {
  fontSize: 16,
  fontStyle: FONT_STYLE_NORMAL,
  fontFamily: FC_SUBJECT_BOLD,
  fontWeight: FONT_WEIGHT_700,
  lineHeight: 22.4,
};

// H4
const Heading4 = {
  fontSize: 15,
  fontStyle: FONT_STYLE_NORMAL,
  fontFamily: FC_SUBJECT_BOLD,
  fontWeight: FONT_WEIGHT_500,
  lineHeight: 21,
};

// H5
const Heading5 = {
  fontSize: 13,
  fontStyle: FONT_STYLE_NORMAL,
  fontFamily: FC_SUBJECT_BOLD,
  fontWeight: FONT_WEIGHT_500,
  lineHeight: 22.4,
};

// H6
const Heading6 = {
  fontSize: 10,
  fontStyle: FONT_STYLE_NORMAL,
  fontFamily: FC_SUBJECT_BOLD,
  fontWeight: FONT_WEIGHT_500,
  lineHeight: 14,
};

// P1
const Paragraph1 = {
  fontSize: 18,
  fontStyle: FONT_STYLE_NORMAL,
  fontFamily: FC_SUBJECT_REGULAR,
  fontWeight: FONT_WEIGHT_400,
  lineHeight: 25.2,
};

// P2
const Paragraph2 = {
  fontSize: 16,
  fontStyle: FONT_STYLE_NORMAL,
  fontFamily: FC_SUBJECT_REGULAR,
  fontWeight: FONT_WEIGHT_500,
  lineHeight: 22.4,
};

const fonts = {
  heading: {
    Heading1,
    Heading2,
    Heading3,
    Heading4,
    Heading5,
    Heading6,
  },
  paragraph: {
    Paragraph1,
    Paragraph2,
  },
  main: {
    FC_SUBJECT_REGULAR,
    FC_SUBJECT_BOLD,
  },
};

export default fonts;
