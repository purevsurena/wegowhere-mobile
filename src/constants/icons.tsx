// direction icons
import { IconTypes } from '@/types/icons';
import addIcon from '@assets/svg/common/add-icon.svg';
import backIcon from '@assets/svg/common/back-icon.svg';

// card icons
import jcb from '@assets/svg/cards/jcb.svg';
import mastercardSecureCode from '@assets/svg/cards/mastercard-securecode.svg';
import mastercard from '@assets/svg/cards/mastercard.svg';
import omiseGrey from '@assets/svg/cards/omise-grey.svg';
import verifiedVisaCard from '@assets/svg/cards/verified-by-visa.svg';
import visaFilled from '@assets/svg/cards/visa-filled.svg';
import visa from '@assets/svg/cards/visa.svg';

const icons: IconTypes = {
  addIcon,
  backIcon,
  jcb,
  mastercardSecureCode,
  mastercard,
  omiseGrey,
  verifiedVisaCard,
  visaFilled,
  visa,
};

export default icons;
