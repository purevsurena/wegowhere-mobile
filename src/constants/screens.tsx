const CARD_LIST = 'CardListScreen';
const CARD_ADD = 'CardAddScreen';

const SCREENS = {
  CARD_LIST,
  CARD_ADD,
};

export { SCREENS };
