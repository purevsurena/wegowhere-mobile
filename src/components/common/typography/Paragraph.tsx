import React from 'react';
import { Text } from 'react-native';
import { ParagraphType, ParagraphDefaultProps } from '@/types/typography';
import Fonts from '@/constants/fonts';
import { Colors } from '@/constants';

function Paragraph({
  color,
  label,
  type,
  customStyle,
  numberOfLines,
  textDecorationLine,
  textAlign,
  children,
}: ParagraphType & typeof ParagraphDefaultProps) {
  const fontVariants = type ? Fonts.paragraph[type] : {};
  return (
    <Text
      numberOfLines={numberOfLines}
      style={[
        {
          ...fontVariants,
          color: Colors.default[color] || Colors.default.dark,
          textDecorationLine,
          textAlign,
          ...customStyle,
        },
      ]}>
      {label}
      {children}
    </Text>
  );
}

Paragraph.defaultProps = ParagraphDefaultProps;

export default Paragraph;
