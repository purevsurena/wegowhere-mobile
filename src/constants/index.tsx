import * as Colors from '@/constants/colors';
import * as Fonts from '@/constants/fonts';

export { Colors, Fonts };
