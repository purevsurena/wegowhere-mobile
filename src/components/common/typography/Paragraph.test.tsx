import React from 'react';
import '@testing-library/jest-native/extend-expect';
import { render } from '@testing-library/react-native';
import Paragraph from '@/components/common/typography/Paragraph';
import { Colors } from '@/constants';

describe('Paragraph component', () => {
  test('renders correctly with specified label and default color', () => {
    const { getByText } = render(
      <Paragraph label="Test Paragraph" type="Paragraph1" />,
    );

    // Assert that the component renders with the correct label
    expect(getByText('Test Paragraph')).toBeTruthy();
  });

  test('renders correctly with specified label and lightGray color', () => {
    const { getByText } = render(
      <Paragraph label="Test Paragraph" color="lightGray" />,
    );

    // Assert that the component renders with the correct label
    expect(getByText('Test Paragraph')).toBeTruthy();

    // Assert that the component applies the specified lightGray color
    expect(getByText('Test Paragraph')).toHaveStyle({
      color: Colors.default.lightGray,
    });

    const { toJSON } = render(
      <Paragraph label="Test Paragraph" color="lightGray" />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
