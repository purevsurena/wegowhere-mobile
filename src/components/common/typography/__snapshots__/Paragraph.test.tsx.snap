// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph component renders correctly with specified label and lightGray color 1`] = `
<Text
  style={
    [
      {
        "color": "#E6E3E6",
        "fontFamily": "FC Subject [Non-commercial] Reg",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 25.2,
        "textAlign": undefined,
        "textDecorationLine": undefined,
      },
    ]
  }
>
  Test Paragraph
</Text>
`;
